main.tf

```diff
@@ -37,6 +37,7 @@ locals {
         ip_name      = "${subnet.name}-${service.service_name}-gateway-${replace(subnet.zone, "/${var.region}-/", "")}-ip"
         subnet_id    = subnet.id
         gateway_name = service.vpe_name != null ? service.vpe_name : "${var.prefix}-${var.vpc_name}-${service.service_name}"
+        name         = service.vpe_name != null ? "${service.vpe_name}-${replace(subnet.zone, "/${var.region}-/", "")}" : "${var.prefix}-${var.vpc_name}-${service.service_name}-${replace(subnet.zone, "/${var.region}-/", "")}"
```

Comment:
What is the difference between `ip_name` and `name`?

Reply:
in terms of why do we need both

Reply:
ip_name is currently used as the `map` index. If we change the map key name, the map entry gets removed and a new one gets created, which results in a destroy / create. If we leave the map key name untouched and just change the name of the resource it results in a update in place.
The `name` is actually based on the `gateway_name` with a -1, -2 or -3 taken from the zone (eg. us-south-1), which both shorter and has a user override in it.